include/seqan3/io/stream/iterator.hpp

```diff
@@ -166,6 +171,124 @@ class fast_istreambuf_iterator
+    ~fast_ostreambuf_iterator() noexcept = default; //!< Defaulted.
+
+    //!\brief Construct from a stream buffer.
+    explicit fast_ostreambuf_iterator(std::basic_streambuf<char_t, traits_t> & ibuf) :
+        stream_buf{reinterpret_cast<stream_buffer_exposer<char_t, traits_t> *>(&ibuf)}
+    {
+        assert(stream_buf != nullptr);
+        if (stream_buf->pptr() == stream_buf->epptr())
+            stream_buf->overflow(); // ensures that put area has space available
+    }
+    //!\}
+
+    /*!\name Arithmetic operators
+     * \{
+     */
+    //!\brief no op.
+    fast_ostreambuf_iterator & operator++()
+    {
+        return *this;
+    }
+    //!\overload
+    fast_ostreambuf_iterator & operator++(int)
+    {
+        return *this;
+    }
+    //!\}
+
+    //!\brief no op.
+    fast_ostreambuf_iterator & operator*()
+    {
+        return *this;
+    }
+
+    //!\brief Writes a character to the associated output stream.
+    fast_ostreambuf_iterator & operator=(char_t const c)
+    {
+        assert(stream_buf != nullptr);
+        if (stream_buf->pptr() == stream_buf->epptr())
+        {
+            if (stream_buf->sputc(c) == traits_t::eof()) // overflow() [virtual], then write character
+                throw std::ios_base::failure{"Cannot write to output stream (reached traits::eof() condition)."};
+        }
+        else
+        {
+            *stream_buf->pptr() = c;
+            stream_buf->pbump(1); // advance pptr() in put area without any checks
+        }
+        return *this;
+    }
+
+    //!\brief Returns `true if this iterator has encountered the end-of-file condition on output, `false` otherwise.
+    bool failed() const noexcept
+    {
+        return stream_buf->overflow() == traits_t::eof();
+    }
+
+    /*!\brief Writes a range to the associated output.
+     * \tparam range_type The type of range to write; Must model std::ranges::forward_range.
+     * \param[in] rng The range to write.
+     */
+    template <std::ranges::forward_range range_type>
+    auto write_range(range_type && rng)
+    {
+        auto buffer_space = stream_buf->epptr() - stream_buf->pptr();
+        auto copy_res = std::ranges::copy(rng | views::take(buffer_space), stream_buf->pptr());
+        auto written_chars = copy_res.out - stream_buf->pptr();
+        stream_buf->pbump(written_chars);
+
+        auto remaining_range = rng | views::drop(written_chars);
+
+        if (remaining_range.begin() == remaining_range.end()) // no more characters to write
+            return copy_res.in;
+
+        // Push one more character and flush
+        if (stream_buf->overflow(*copy_res.in) == traits_t::eof())
+            throw std::ios_base::failure{"Cannot write to output stream (reached traits::eof() condition)."};
+
+        // drop 1 character that has been written in overflow() and write remaining range recursively
+        return write_range(remaining_range | views::drop(1));
```

Comment:
```suggestion
        return write_range(remaining_range | views::drop(written_chars + 1));
```

The previous small changes prevent creating two drop-views in case no flattening happens.